internal/cloud/cloud.go

```diff
@@ -101,26 +104,24 @@ func getVirtualMachine(username string) string {
 	if envGateway := os.Getenv("DEVBOX_GATEWAY"); envGateway != "" {
 		client.Hostname = envGateway
 	}
-	bytes, err := client.Exec("auth")
+	sshOut, err := client.Exec("auth")
 	if err != nil {
-		log.Println(err)
-		var exitErr *exec.ExitError
-		if errors.As(err, &exitErr) {
-			log.Printf("ssh %s stderr:\n%s", client.Hostname, string(exitErr.Stderr))
-		}
-		os.Exit(1)
-	}
-	debug.Log("ssh %s stdout:\n%s", client.Hostname, string(bytes))
+		log.Fatalln("error requesting VM:", err)
+	}
 	resp := &vm{}
-	err = json.Unmarshal(bytes, resp)
-	if err != nil {
-		log.Fatal(err)
+	if err := json.Unmarshal(sshOut, resp); err != nil {
+		log.Fatalf("error unmarshaling gateway response %q: %v", sshOut, err)
 	}
-	if resp.VMPrivateKey != "" {
-		err := openssh.AddVMKey(resp.VMHost, resp.VMPrivateKey)
-		if err != nil {
-			log.Fatal(err)
-		}
+	if redacted, err := json.MarshalIndent(resp.redact(), "\t", "  "); err == nil {
+		debug.Log("got gateway response:\n\t%s", redacted)
+	}
+	if resp.VMPrivateKey == "" {
```

Comment:
what are the scenarios for when VMPrivateKey is empty versus filled?

Reply:
It's filled the first time it's created. After that, it throws away its copy of the private key and expects the client to already have it.